app/src/main/java/edu/nextstep/camp/calculator/MainActivity.kt

```diff
@@ -3,20 +3,28 @@ package edu.nextstep.camp.calculator
 import android.os.Bundle
 import android.widget.Toast
 import androidx.appcompat.app.AppCompatActivity
+import androidx.core.view.isGone
+import androidx.core.view.isVisible
 import edu.nextstep.camp.calculator.databinding.ActivityMainBinding
-import edu.nextstep.camp.domain.Calculator
-import edu.nextstep.camp.domain.Expression
-import edu.nextstep.camp.domain.Operator
+import edu.nextstep.camp.domain.*
 
 class MainActivity : AppCompatActivity(), MainContract.View {
     private lateinit var binding: ActivityMainBinding
-    private val presenter: MainContract.Presenter = MainPresenter(this, Calculator())
+    private lateinit var calculationHistoryAdapter: CalculationHistoryAdapter
```

Comment:
```suggestion
    private val calculationHistoryAdapter: CalculationHistoryAdapter = CalculationHistoryAdapter()
```
지연 초기화는 주로 라이프싸이클에 맞춰서 객체가 초기화되야하는 경우에 주로 사용하는 것이 좋아요.
어뎁터는 라이프싸이클과 무관하게 미리 선언해두고 사용할 수 있어요.
지연 초기화가 아닌 일반 초기화를 활용해보면 어떨까요?